src/beacon.js

```diff
@@ -35,6 +34,13 @@ export default function Emitter(endpoint) {
     }
   }
 
+  function wrapMessages(msgs) {
+    return {
+      uid: context.uid,
```

Comment:
This should be the responsibility of the code emitting the beacon. This is making the interface leaky.